scripts/tests/backwards-compatibility-test.sh

```diff
@@ -11,6 +11,9 @@ versions=( "${@:-${default_versions[@]}}" )
 
 # signal to downstream test scripts
 export FM_BACKWARDS_COMPATIBILITY_TEST=1
+# previous versions did not have unknown module so can't support it,
```

Comment:
@bradleystachurski Food for thought: we will need to deal with all sorts of incompatibilities between versions.

Reply:
Yeah, I'd anticipate we'll have a lot of conditional logic in devimint based on versions similar to [`18f00a8` (#4247)](https://github.com/fedimint/fedimint/pull/4247/commits/18f00a86e731a1c037f2345b7f1e3aa9890022c0#diff-75cef4315751be4fa6703860684a38e9c382d522c05f6df275846e94a373562dR102)

Reply:
It might be useful to come up with a system that'll allow us to easily cleanup conditional logic that's no longer needed in devimint.

e.g. `// FIXME: deprecated v0.2.1` so we can grep for a version when it's reached EOL

Reply:
:brain: But we should make it an issue and brainstorm a little.

Reply:
👍 https://github.com/fedimint/fedimint/issues/4424